index.js

```diff
@@ -253,16 +256,46 @@ module.exports = function AsyncAppendOnlyLog(filename, opts) {
     getBlock(offset, function gotBlockForDelete(err, blockBuf) {
```

Comment:
There is a potential minor problem in this one. If you delete something within the same block and the scheduled delete has not been flushed and you somehow managed to not have the block in cache anymore (getBlock uses the cache), then you could get a block from disk that does not have the first delete and thus you'll miss one or more deletes. A workaround is to check blocksWithDeletables and use that to get the block if it is there. As said, this is highly unlikely to happen so I would probably be fine with leaving it as is.

Reply:
Well spotted! Covering this case is easy